synapse/config/repository.py

```diff
@@ -181,11 +187,17 @@ def read_config(self, config, **kwargs):
                 )
 
             if "url_preview_ip_range_blacklist" not in config:
-                raise ConfigError(
-                    "For security, you must specify an explicit target IP address "
-                    "blacklist in url_preview_ip_range_blacklist for url previewing "
-                    "to work"
-                )
+                if "HTTP_PROXY" not in os.environ and "HTTPS_PROXY" not in os.environ:
+                    raise ConfigError(
+                        "For security, you must specify an explicit target IP address "
+                        "blacklist in url_preview_ip_range_blacklist for url previewing "
+                        "to work"
+                    )
+                else:
+                    proxy_var = (
+                        "HTTP_PROXY" if "HTTP_PROXY" in os.environ else "HTTPS_PROXY"
+                    )
+                    logger.warning("".join(HTTP_PROXY_SET_WARNING % proxy_var))
```

Comment:
this warning will be written when a `url_preview_ip_range_blacklist` is *not* set. That seems surprising?

Reply:
Ah. Thank you for the catch. Will go ahead and edit this so the warning is written when `url_preview_ip_range_blacklist` _is_ set AND a proxy is set.  

In addition to this, would it be beneficial to have a separate info log generated when `url_preview_ip_range_blacklist` _is not_ set AND a proxy is set?

Reply:
> In addition to this, would it be beneficial to have a separate info log generated when `url_preview_ip_range_blacklist` _is not_ set AND a proxy is set?

I don't really think that is necessary?